dashboard/app/models/aidiff_thread.rb

```diff
@@ -25,4 +25,23 @@ class AidiffThread < ApplicationRecord
   has_many :aidiff_messages
 
   validates :context_type, inclusion: {in: SharedConstants::AI_DIFF_CONTEXT.values}
+
+  def summarize
+    {
+      id: id,
+      title: title,
+      updated_at: updated_at,
+      context_type: context_type,
+    }
+  end
+
+  def summarize_with_messages
+    {
+      id: id,
+      title: title,
+      updated_at: updated_at,
+      context_type: context_type,
+      messages: aidiff_messages&.map(&:summarize)
+    }
```

Comment:
nit: makes a bit clearer that this is supposed to be a superset of `summarize`

```suggestion
    summarize.merge(
      {
        messages: aidiff_messages&.map(&:summarize)
      }
    )
```